src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.FruitTransaction.Operation;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+
+        for (int i = 1; i < data.size(); i++) {
+            String line = data.get(i);
+            String[] parts = line.split(",");
+
+            if (parts.length < 3) {
+                throw new IllegalArgumentException("Invalid data format at line "
+                        + (i + 1) + ": " + line);

```

Comment:
The check for the parts length is good for validation, but you might want to encapsulate this logic into a separate validation method to keep the `convertToTransaction` method focused on conversion only.